### ISSUE TITLE
Crash if Mod Menu is not installed on Fabric

### ISSUE THREAD

**Issue by jepotter1-archive** [yungnickyoung/Better-Caves#118]
It would be helpful if YUNG's Better Caves were to depend on Mod Menu optionally rather than making it a required dependency since many Fabric users do not use Mod Menu.

**Comment by jepotter1-archive** [comment]
This is likely where the crash originates from:

https://github.com/yungnickyoung/YUNGs-Better-Caves/blob/03f3fb107ea7c116ff8c05060fbc5c8b97518558/src/main/java/com/yungnickyoung/minecraft/bettercaves/config/gui/BCModMenu.java#L5

**Comment by jwinnie** [comment]
Crash report:

```
java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'modmenu'!
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:53)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:36)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointClient.start(EntrypointClient.java:33)
	at net.minecraft.class_310.<init>(class_310.java:437)
	at net.minecraft.client.main.Main.main(Main.java:177)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
Caused by: net.fabricmc.loader.api.EntrypointException: Exception while loading entries for entrypoint 'modmenu' provided by 'bettercaves'
	at net.fabricmc.loader.EntrypointStorage.lambda$getEntrypointContainers$1(EntrypointStorage.java:186)
	at net.fabricmc.loader.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:37)
	at io.github.prospector.modmenu.ModMenu.lambda$onInitializeClient$0(ModMenu.java:75)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at io.github.prospector.modmenu.ModMenu.onInitializeClient(ModMenu.java:74)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:50)
	... 11 more
Caused by: net.fabricmc.loader.api.LanguageAdapterException: java.lang.ClassNotFoundException: com.yungnickyoung.minecraft.bettercaves.config.gui.BCModMenu
	at net.fabricmc.loader.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:47)
	at net.fabricmc.loader.EntrypointStorage$NewEntry.create(EntrypointStorage.java:112)
	at net.fabricmc.loader.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:99)
	at net.fabricmc.loader.EntrypointStorage.lambda$getEntrypointContainers$1(EntrypointStorage.java:184)
	... 16 more
Caused by: java.lang.ClassNotFoundException: com.yungnickyoung.minecraft.bettercaves.config.gui.BCModMenu
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:468)
	at net.fabricmc.loader.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:45)
	... 19 more
```

**Comment by yungnickyoung** [comment]
Yeah, I agree. I regret making this decision when I first ported. It's intended to be an embedded dependency, actually, but I'm guessing ModMenu has updated enough for it to be broken?

Anyway, it will be marked as an optional dependency in the future. I can't give a target date though, I've got a lot of other things on my plate atm.

**Comment by ghost** [comment]
Would also greatly recommend making it an optional dependency, spent some time trying to solve where that crash is comming from. It told me something about Mod Menu and I was confused since I did not even use Mod Menu.

**Comment by yungnickyoung** [comment]
Fixed in v1.2. Please open a new issue if a problem like this persists.
